File: Lambda/Batch-Layer/src/main/scala/prv/zielony/transaction/analyst/lambda/batch/layer/events/transactions/TransferEvent.scala
package prv.zielony.transaction.analyst.lambda.batch.layer.events.transactions

import java.time.LocalDateTime

import prv.zielony.transaction.analyst.lambda.model.Currency

/**
 * Created by zielony on 14.02.16.
 */
case class TransferEvent(override val timestamp:LocalDateTime, val transactionId:Long,
                         override val status:TransactionStatus, val sourceAccountId:Long, val targetAccountId:Long,
                         val amount:Double, val currency: Currency) extends TransactionEvent(timestamp, transactionId, status);

File: Classical/src/main/java/prv/zielony/transaction/analyst/classical/dto/TransferEvent.java
package prv.zielony.transaction.analyst.classical.dto;

import prv.zielony.transaction.analyst.classical.model.Currency;

import java.time.LocalDateTime;

/**
 * Created by zielony on 27.02.16.
 */
public class TransferEvent {

    private LocalDateTime timestamp;

    private Long sourceAccountId;

    private Long targetAccountId;

    private Double amount;

    public Long getSourceAccountId() {
        return sourceAccountId;
    }

    public void setSourceAccountId(Long sourceAccountId) {
        this.sourceAccountId = sourceAccountId;
    }

    public LocalDateTime getTimestamp() {
        return timestamp;
    }

    public void setTimestamp(LocalDateTime timestamp) {
        this.timestamp = timestamp;
    }

    public Long getTargetAccountId() {
        return targetAccountId;
    }

    public void setTargetAccountId(Long targetAccountId) {
        this.targetAccountId = targetAccountId;
    }

    public Double getAmount() {
        return amount;
    }

    public void setAmount(Double amount) {
        this.amount = amount;
    }
}